pkg/agent/apiserver/handlers/podinterface/handler_test.go

```diff
@@ -0,0 +1,237 @@
+			ContainerID:  "containerid0",
+			PodName:      "pod0",
+			PodNamespace: "namespace0",
+		},
+	},
+	{
+		InterfaceName: "interface1",
+		IP:            net.ParseIP(ipStrs[1]),
+		MAC:           macs[1],
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "portuuid1",
+			OFPort:   1,
+		},
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+			ContainerID:  "containerid1",
+			PodName:      "pod1",
+			PodNamespace: "namespace1",
+		},
+	},
+	{
+		InterfaceName: "interface2",
+		IP:            net.ParseIP(ipStrs[2]),
+		MAC:           macs[2],
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: "portuuid2",
+			OFPort:   2,
+		},
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+			ContainerID:  "containerid2",
+			PodName:      "pod2",
+			PodNamespace: "namespace0",
+		},
+	},
+}
+
+func parseMAC(mac string) net.HardwareAddr {
+	res, _ := net.ParseMAC(mac)
+	return res
+}
+
+func TestPodInterfaceQuery(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	testcases := map[string]struct {
+		query           string
+		expectedStatus  int
+		expectedContent Response
+	}{
+		"Hit pod interface query": {
+			query:           "?name=pod0&&namespace=namespace0&&all-namespaces=false",
+			expectedStatus:  http.StatusOK,
+			expectedContent: responses[0],
+		},
+		"Miss pod interface query": {
+			query:           "?name=pod1&&namespace=namespace0&&all-namespaces=false",
+			expectedStatus:  http.StatusNotFound,
+			expectedContent: Response{},
+		},
+	}
+
+	for k, tc := range testcases {
+		// mock GetInterfacesByType() in pkg interfacestore
+		i := mock_interfacestore.NewMockInterfaceStore(ctrl)
+		i.EXPECT().GetInterfacesByType(interfacestore.ContainerInterface).Return(testInterfaceConfigs).AnyTimes()
+
+		// mock GetInterfaceStore() in pkg monitor
+		q := monitor.NewMockAgentQuerier(ctrl)
+		q.EXPECT().GetInterfaceStore().Return(i).AnyTimes()
+		handler := HandleFunc(q)
+
+		req, err := http.NewRequest(http.MethodGet, tc.query, nil)
+		assert.Nil(t, err)
+
+		recorder := httptest.NewRecorder()
+		handler.ServeHTTP(recorder, req)
+		assert.Equal(t, tc.expectedStatus, recorder.Code, k)
+
+		if tc.expectedStatus == http.StatusOK {
+			var received Response
+			err = json.Unmarshal(recorder.Body.Bytes(), &received)
+			assert.Nil(t, err)
+			assert.Equal(t, tc.expectedContent, received)
+		}
+	}
+}
+
+func TestPodInterfaceListQuery(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	testcases := map[string]struct {
+		query           string
+		expectedStatus  int
+		expectedContent []Response
+	}{
+		"Hit pod interface list query": {
+			query:           "?name=&&namespace=namespace0&&all-namespaces=false",
+			expectedStatus:  http.StatusOK,
+			expectedContent: []Response{responses[0], responses[2]},
+		},
+		"Miss pod interface list query": {
+			query:           "?name=pod1&&namespace=namespace0&&all-namespaces=false",
```

Comment:
This is not a list query but a get query duplicated with `TestPodInterfaceQuery`

Reply:
Removed